backend/typescript/services/implementations/campService.ts

```diff
@@ -1,17 +1,93 @@
-import { CreateCampDTO, CamperCSVInfoDTO, CampDTO } from "../../types";
+import {
+  CreateCampDTO,
+  CampDTO,
+  CamperCSVInfoDTO,
+  GetCampDTO,
+  FormQuestionDTO,
+} from "../../types";
 import ICampService from "../interfaces/campService";
-import MgCampSession, { CampSession } from "../../models/campSession.model";
-import MgCamper, { Camper } from "../../models/camper.model";
 import { getErrorMessage } from "../../utilities/errorUtils";
 import { generateCSV } from "../../utilities/CSVUtils";
 import logger from "../../utilities/logger";
 import MgCamp, { Camp } from "../../models/camp.model";
-import MgFormQuestion from "../../models/formQuestion.model";
+import MgCampSession, { CampSession } from "../../models/campSession.model";
+import MgFormQuestion, { FormQuestion } from "../../models/formQuestion.model";
+import MgCamper, { Camper } from "../../models/camper.model";
 
 const Logger = logger(__filename);
 
 class CampService implements ICampService {
   /* eslint-disable class-methods-use-this */
+  async getCamps(): Promise<GetCampDTO[]> {
+    try {
+      const camps: Camp[] | null = await MgCamp.find({})
+        .populate({
+          path: "campSessions",
+          model: MgCampSession,
+          populate: {
+            path: "campers",
+            model: MgCamper,
+          },
+        })
+        .populate({
+          path: "formQuestions",
+          model: MgFormQuestion,
+        });
+
+      if (!camps) {
+        return [];
+      }
+
+      return camps.map((camp) => {
+        const formQuestionArr = camp.formQuestions as FormQuestion[];
+        const formQuestions = formQuestionArr.map(
```

Comment:
nit: do you need the formQuestionsArr or can you do this?
```suggestion
        const formQuestions = (camp.formQuestions as FormQuestion[]).map(

```